easybuild/easyblocks/x/xmipp.py

```diff
@@ -0,0 +1,161 @@
+##
+# Copyright 2015-2015 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Xmipp, implemented as an easyblock
+
+@author: Jens Timmerman (Ghent University)
+"""
+import glob
+import os
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.filetools import mkdir, extract_file
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+
+from easybuild.easyblocks.generic.pythonpackage import det_pylibdir
+
+
+class EB_Xmipp(EasyBlock):
+    """Support for building/installing Xmipp."""
+
+    def __init__(self, *args, **kwargs):
+        """Easyblock constructor, enable building in installation directory."""
+        super(EB_Xmipp, self).__init__(*args, **kwargs)
+        self.build_in_installdir = True
+
+    def extract_step(self):
+        """Extract sources."""
+        # strip off 'xmipp' part to avoid having everything in a 'xmipp' subdirectory
+        if not self.cfg['unpack_options']:
+            self.cfg['unpack_options'] = '--strip-components=1'
```

Comment:
no pls, you really need the `--strip-components=1` bit below (and you're not sure it'll be there if `unpack_options` is defined by the easyconfig), so use this:

```python
self.cfg.update('unpack_options', '--strip-components=1')
```

I'll merge https://github.com/hpcugent/easybuild-framework/pull/1229 as soon as Jenkins gives it a green light, no worries.